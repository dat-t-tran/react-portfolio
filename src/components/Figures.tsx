import React from 'react'
import { classNames } from '../utils/utils';

interface FiguresProps {
    gap: string;
    description: React.ReactNode;
    children: React.ReactNode;
    direction?: string;
    customClass?: string;
}

const Figures: React.FC<FiguresProps> = ({ gap, description, children, customClass = "", direction = "" }) => {
    return (
        <div className={classNames("inline-flex py-2 px-4 sm:p-4 items-center rounded-[0.625rem] sm:rounded-[1.25rem] border border-[#ECF4FD] bg-white absolute", gap, direction, customClass)}>
            {children}
            <div className='text-secondary text-center text-[0.6875rem] sm:text-[1.125rem] font-normal leading-[1.03125rem] sm:leading-[1.6875rem] -tracking-[0.01375rem] sm:-tracking-[0.0225rem]'>{description}</div>
        </div>
    )
}

export default Figures