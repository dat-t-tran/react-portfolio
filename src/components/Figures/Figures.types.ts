export interface FiguresProps {
  gap: string;
  description: React.ReactNode;
  children: React.ReactNode;
  rounded?: boolean;
  direction?: string;
  customClass?: string;
}
