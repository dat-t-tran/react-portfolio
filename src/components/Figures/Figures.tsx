import React, { useMemo } from 'react';
import { classNames } from '../../utils/classNames';
import { FiguresProps } from './Figures.types';

const Figures: React.FC<FiguresProps> = React.memo(
  ({
    gap,
    description,
    children,
    customClass = '',
    direction = '',
  }): JSX.Element => {
    const containerClassNames = useMemo(() => {
      return classNames(
        'inline-flex py-2 px-4 sm:p-4 items-center rounded-[0.625rem] sm:rounded-[1.25rem] border border-[#ECF4FD] bg-white absolute',
        gap,
        direction,
        customClass
      );
    }, [gap, direction, customClass]);

    return (
      <div className={containerClassNames}>
        {children}
        <div className="text-secondary text-center text-[0.6875rem] sm:text-[1.125rem] font-normal leading-[1.03125rem] sm:leading-[1.6875rem] -tracking-[0.01375rem] sm:-tracking-[0.0225rem]">
          {description}
        </div>
      </div>
    );
  }
);

export default Figures;
