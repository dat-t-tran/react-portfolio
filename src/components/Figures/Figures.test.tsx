import { render } from '@testing-library/react';
import Figures from './Figures';
import { describe, it, expect } from 'vitest';

describe('Figures Component', () => {
    it('renders correctly with required props', () => {
        const { container } = render(
            <Figures gap="gap-4" description="Test Description">
                <div>Child Element</div>
            </Figures>
        );
        expect(container).toMatchSnapshot();
    });

    it('applies customClass correctly', () => {
        const { container } = render(
            <Figures gap="gap-4" description="Test Description" customClass="custom-class">
                <div>Child Element</div>
            </Figures>
        );
        expect(container.firstChild).toHaveClass('custom-class');
    });

    it('applies direction correctly', () => {
        const { container } = render(
            <Figures gap="gap-4" description="Test Description" direction="flex-row">
                <div>Child Element</div>
            </Figures>
        );
        expect(container.firstChild).toHaveClass('flex-row');
    });

    it('renders children correctly', () => {
        const { getByText } = render(
            <Figures gap="gap-4" description="Test Description">
                <div>Child Element</div>
            </Figures>
        );
        expect(getByText('Child Element')).toBeInTheDocument();
    });

    it('renders description correctly', () => {
        const { getByText } = render(
            <Figures gap="gap-4" description="Test Description">
                <div>Child Element</div>
            </Figures>
        );
        expect(getByText('Test Description')).toBeInTheDocument();
    });
});