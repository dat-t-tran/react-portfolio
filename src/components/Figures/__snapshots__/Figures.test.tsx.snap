// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Figures Component > renders correctly with required props 1`] = `
<div>
  <div
    class="rounded-[0.625rem] sm:rounded-[1.25rem] border border-[#ECF4FD] inline-flex items-center bg-white gap-4"
  >
    <div>
      Child Element
    </div>
    <div
      class="text-secondary text-center text-[0.6875rem] sm:text-[1.125rem] font-normal leading-[1.03125rem] sm:leading-[1.6875rem] -tracking-[0.01375rem] sm:-tracking-[0.0225rem]"
    >
      Test Description
    </div>
  </div>
</div>
`;
